### PR TITLE
fix: improve  `<Radio />` in height contrast style

### LINKED ISSUE #41821: Radio button support for high contrast
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/radio-button-high-contrast-support-srmd5m?file=/index.js)

### Steps to reproduce

1. Go to [Radio](https://ant.design/components/radio) component in the Ant Design documentation.
2. Activate high contrast mode. You can emulate high contrast in Google Chrome with DevTools.

DevTools -> Rendering -> Emulate CSS media feature forced-colors -> forced-colors: active

3. Look at the Radio button examples.

### What is expected?

On high contrast mode, should be able to differentiate the states of a radio button.

### What is actually happening?

Cannot visually differentiate checked from unchecked, or disabled states.

| | Default | forced-colors: active | forced-colors: active + prefers-color-scheme: dark |
| ---- | ---- | ---- | ---- |
| Default | Screenshot 2023-04-14 at 15 34 25 | Screenshot 2023-04-14 at 15 38 00 | Screenshot 2023-04-14 at 15 38 14 |
| Disabled | Screenshot 2023-04-14 at 15 34 19 | Screenshot 2023-04-14 at 15 37 54 | Screenshot 2023-04-14 at 15 38 10 | 

| Environment | Info |
| --- | --- |
| antd | 5.4.2 |
| React | react |
| System | MacBook Air M2 |
| Browser | Goggle Chrome Version 110.0.5481.100 (Official Build) (arm64)  |

---
In the reproduction link example there is an example solution combining [`<system-color>`](https://developer.mozilla.org/en-US/docs/Web/CSS/system-color) CSS data type and [`forced-colors`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors) media query.

For context, from the article [Styling for Windows high contrast with new standards for forced colors](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/): 
> High contrast on Windows is an accessibility feature designed to increase text legibility and improve readability. The feature works by enabling the user to select theme colors for a scoped number of semantic elements.

The Radio component uses a visually hidden radio input element and then renders the radio button with span elements. The user agent does not correctly override the colors for this span structure since it's not using semantic elements. 

The example solution shows how one could, while in high contrast mode, implement these user agent color overrides manually for the Radio component.

Example fix screenshots and native elements for reference:

`forced-colors: active` and `prefers-color-scheme: light`
Screenshot 2023-04-14 at 15 50 27
Screenshot 2023-04-14 at 15 45 47

`forced-colors: active` and `prefers-color-scheme: dark`
Screenshot 2023-04-14 at 15 50 18
Screenshot 2023-04-14 at 15 45 59

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #41821

### 💡 Background and solution

**Before**
| | Default | forced-colors: active | forced-colors: active + prefers-color-scheme: dark |
| ---- | ---- | ---- | ---- |
| Default | Screenshot 2023-04-14 at 15 34 25 | Screenshot 2023-04-14 at 15 38 00 | Screenshot 2023-04-14 at 15 38 14 |
| Disabled | Screenshot 2023-04-14 at 15 34 19 | Screenshot 2023-04-14 at 15 37 54 | Screenshot 2023-04-14 at 15 38 10 | 

**After**
| | Default | forced-colors: active | forced-colors: active + prefers-color-scheme: dark |
| ---- | ---- | ---- | ---- |
| Default |Screenshot 2023-04-14 at 15 34 25  | image | image |
| Disabled |Screenshot 2023-04-14 at 15 34 19 | image | image |

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      improve  `<Radio />` in height contrast style     |
| 🇨🇳 Chinese |    改进 `<Radio />` 在高度对比下样式       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 1e8fe34</samp>

This pull request enhances the radio components in the `ant-design` library by adding a function to support high contrast mode and an example to show the solid button style. It modifies the files `components/radio/style/index.tsx` and `components/radio/demo/radiobutton-solid.tsx`.

### <samp>🤖 Generated by Copilot at 1e8fe34</samp>

*  Add a new example of using solid radio buttons ([link](https://github.com/ant-design/ant-design/pull/41933/files?diff=unified&w=0#diff-dbacf2ef51ce02fee91ff838aab3d0a0bc8d8c60be1e670fa5d81d790c9e491cR20-R29))
*  Improve the accessibility of radio components in high contrast mode ([link](https://github.com/ant-design/ant-design/pull/41933/files?diff=unified&w=0#diff-a741439225c8413e967262511dad718192a7a5154ee037fa3c679681b95372afR482-R555), [link](https://github.com/ant-design/ant-design/pull/41933/files?diff=unified&w=0#diff-a741439225c8413e967262511dad718192a7a5154ee037fa3c679681b95372afR618))

### CODE REVIEW

File: components/radio/__tests__/__snapshots__/demo.test.tsx.snap [hunk 1]
```diff
@@ -331,13 +331,12 @@ Array [
     style="margin-top:16px"
   >
     <label
-      class="ant-radio-button-wrapper ant-radio-button-wrapper-checked ant-radio-button-wrapper-disabled"
+      class="ant-radio-button-wrapper ant-radio-button-wrapper-disabled"
```
Comment (zombieJ) [inline components/radio/__tests__/__snapshots__/demo.test.tsx.snap:334]: 好奇怪，为什么这里会少cls？

**Comment by zombieJ** [comment]
可以挪到 feature br，然后类似主题切换加个高对比切换这样方便看

File: components/radio/__tests__/__snapshots__/demo.test.tsx.snap (hunk 1 above)
Comment (Wxh16144) [reply components/radio/__tests__/__snapshots__/demo.test.tsx.snap:334]: 我改了一下 solid 模式下的 Radio, 想看一下 禁用状态 and  选中状态 = 最终是啥情况. 其实如果没必要加, 我可以把这个 revert 掉. 只做和高对比度有关的提交.

**Comment by Wxh16144** [comment]
> 类似主题切换加个高对比

这个怎么实现呢? 然后其实我也有在想, 后面是不是可以和设计师讨论一下出一个高度对比度的算法? 和 dark 算法一样, 这样或许这个提交就必要了, 当然不知道有没有可能实现, 或者实现难度

**Comment by yoyo837** [comment]
有一个相同目的PR #42986

File: components/radio/style/index.tsx [hunk 2]
```diff
@@ -479,6 +479,80 @@ const getRadioButtonStyle: GenerateStyle<RadioToken> = (token) => {
...
+          },
+
+          '&:not(:first-child)::before': {
+            backgroundColor: 'initial',
+            transition: 'none',
+          },
+
+          [`&-checked:not(${componentCls}-button-wrapper-disabled)`]: {
+            color: 'Highlight',
+            borderColor: 'Highlight',
```
Comment (daryazata) [inline components/radio/style/index.tsx:520]: im wondering, isnt there a way to put the string "Highlight" into an enum, otherfiles it feels like error prone..

Comment (Wxh16144) [reply components/radio/style/index.tsx:520]: I don't know the solution yet, need help from core members. 😭

**Comment by Wxh16144** [comment]
The solution is not very easy to maintain later, more discussion is needed.

**Pull request closed without merging**